src/orders/components/OrderTransaction/components/CardTitle/CardTitle.test.tsx

```diff
@@ -0,0 +1,160 @@
+import { TransactionItemFragment } from "@dashboard/graphql";
+import { orderTransactions } from "@dashboard/orders/fixtures";
+import Wrapper from "@test/wrapper";
+import { render, screen } from "@testing-library/react";
+import * as React from "react";
+
+import { CardTitleProps, OrderTransactionCardTitle } from "./CardTitle";
+
+describe("OrderTransactionCardTitle", () => {
+  const transaction = orderTransactions[0] as TransactionItemFragment;
+  const cardTitleProps: CardTitleProps = {
+    transaction,
+    onTransactionAction: jest.fn(),
+    idx: 1,
+    showActions: true,
+  };
+
+  it("renders", () => {
+    // Arrange & Act
+    render(
+      <Wrapper>
+        <OrderTransactionCardTitle {...cardTitleProps} />
+      </Wrapper>,
+    );
+
+    // Assert
+    const title = screen.getByTestId("order-transaction-card-title");
+
+    expect(title).toBeInTheDocument();
+  });
+
+  it("shows correct title: createdBy, pspReference, externalUrl", () => {
+    // Arrange & Act
+    render(
+      <Wrapper>
+        <OrderTransactionCardTitle
+          {...cardTitleProps}
+          transaction={{
+            ...transaction,
+            externalUrl: "https://example.com",
+          }}
+        />
+      </Wrapper>,
+    );
+
+    // Assert
+    expect(screen.getByText(`Jacob Smith: ${transaction.name}:`)).toBeInTheDocument();
+    expect(screen.getByText(transaction.pspReference)).toBeInTheDocument();
+    expect(screen.getByTestId("psp-reference")).toBeInTheDocument();
+    expect(screen.getByRole("link")).toHaveAttribute("href", "https://example.com");
+  });
+
+  it("shows correct title: createdBy, pspReference", () => {
+    // Arrange & Act
+    render(
+      <Wrapper>
+        <OrderTransactionCardTitle {...cardTitleProps} />
+      </Wrapper>,
+    );
+
+    // Assert
+    expect(screen.getByText(`Jacob Smith: ${transaction.name}:`)).toBeInTheDocument();
+    expect(screen.getByText(transaction.pspReference)).toBeInTheDocument();
+    expect(screen.getByTestId("psp-reference")).toBeInTheDocument();
+    expect(() => screen.getByTestId("psp-reference-link")).toThrow();
```

Comment:
suggestion: Test should check one thing and do it only once, in this case  `expect(screen.getByText(`Jacob Smith: ${transaction.name}:`)).toBeInTheDocument();` is checking multiple time and cause multiple test to fail. This should be check only once and when fail we will  know what exactly failed.

Reply:
This is intentional. Depending on how much information we have about transaction the title is rendered differently. Since the logic a bit confusing I wanted the test to be very explicit. Our staging db doesn't cover all the cases and I think it will be easy to accidentally change the logic here without being aware of it.